example-get-started-experiments/code/.github/workflows/deploy-model.yml

```diff
@@ -0,0 +1,58 @@
+name: Deploy model
+
+on:
+  push:
+    # When a new version is registered in Studio Model Registry
+    tags:
+      - "results/train=pool-segmentation@*"
+
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Manual version name'
+        required: true
+        type: string
+
+jobs:
+  deploy-model:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - id: clean_version_name
+      run: echo "version=$(echo ${{ github.ref_name || github.event.inputs.version }} | sed 's/[^a-zA-Z0-9-]/-/g')" >> "$GITHUB_OUTPUT"
+
+    - uses: aws-actions/configure-aws-credentials@v2
+      with:
+        aws-region: us-east-1
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.8'
+        cache: 'pip'
+        cache-dependency-path: requirements.txt
+
+    - run: pip install -r requirements.txt
+
+    - run: dvc pull results/train/artifacts/best.pt.dvc
+
+    - env:
+        VERSION: ${{steps.clean_version_name.outputs.version}}
+      run: |
+        bash sagemaker/bundle_and_upload_model.sh \
+        results/train/artifacts/best.pt \
+        s3://pool-segmentation/$VERSION/model.tar.gz
+
+    - env:
+        VERSION: ${{steps.clean_version_name.outputs.version}}
+      run: |
+        python sagemaker/deploy_model.py \
+        --name $VERSION \
+        --model_data s3://pool-segmentation/$VERSION/model.tar.gz \
+        --role ${{ secrets.AWS_ROLE_TO_ASSUME }} \
+        --instance_type 'ml.c4.xlarge'
```

Comment:
This is option  `A`  of  `How to create endpoints based on MR events`.

Explained here : https://www.notion.so/iterative/Sagemaker-Deployment-fce94fa0d81c44f395c10fc32940ab82#f3773819ebce4659ad397ed92686d56c